arduino/smartcar/smartcar.ino

```diff
@@ -1,116 +1,175 @@
+      car.disableCruiseControl();
+        int speed = smartcarlib::utils::getAbsolute(40);
+        targetDegrees %= 360; /* puts it on a (-360,360) scale */
+        
+        if (!targetDegrees)
+            return; /* if the target degrees is 0, don't bother doing anything */
+        
+        /* Let's set opposite speed for the motors on opposite sides, so it rotates on spot */
+        if (targetDegrees > 0) { /* positive value means we should rotate clockwise */
+            car.overrideMotorSpeed(40, -40); /* left motors spin forward, right motors spin backward */
+        } else { /* rotate counter clockwise */
+            car.overrideMotorSpeed(-40,40); /* left motors spin backward, right motors spin forward */
+        }
+        
+        const auto initialHeading = car.getHeading(); /* the initial heading we'll use as offset to calculate the absolute displacement */
+        int degreesTurnedSoFar = 0; /* this variable will hold the absolute displacement from the beginning of the rotation */
+        
+        while (abs(degreesTurnedSoFar) < abs(targetDegrees)) { /* while absolute displacement hasn't reached the (absolute) target, keep turning */
+            car.update(); /* update to integrate the latest heading sensor readings */
+            auto currentHeading = car.getHeading(); /* in the scale of 0 to 360 */
+            
+            if ((targetDegrees < 0) && (currentHeading > initialHeading)) { /* if we are turning left and the current heading is larger than the initial one 
+                                                                            (e.g. started at 10 degrees and now we are at 350), we need to substract 360, 
+                                                                            so to eventually get a signed displacement from the initial heading (-20) */
+                currentHeading -= 360; 
+            } else if ((targetDegrees > 0) && (currentHeading < initialHeading)) { /* if we are turning right and the heading is smaller than the initial one 
+                                                                                   (e.g. started at 350 degrees and now we are at 20), 
+                                                                                   so to get a signed displacement (+30) */
+                currentHeading += 360;
+            }
+            
+            degreesTurnedSoFar = initialHeading - currentHeading; /* degrees turned so far is initial heading minus current 
+                                                                  (initial heading is at least 0 and at most 360. To handle the "edge" cases we substracted or added 
+                                                                  360 to currentHeading) */
+        }
+        car.setSpeed(0); /* we have reached the target, so stop the car */
+    
+       // car.enableCruiseControl();
+    }
+    
+    int measureDistance() {
+    
+        return front.getDistance();
+    }
```

Comment:
functions like this make it too opaque on what its actually doing; since its calling a single member function of a global why not just call it directly? at least name the function appropriately e.g: `measureFrontDistance`